File: app/src/main/java/com/android/mathias/velocity/Route.java
package com.android.mathias.velocity;

import android.content.Context;
import android.location.Location;

import com.google.android.gms.maps.model.LatLng;

import java.util.List;

class Route {
    private long mId;
    private int mPos;
    private String mName;
    private LatLng mStartLoc;
    private LatLng mEndLoc;
    private String mStartName;
    private String mEndName;

    public Route() { }

    public Route(String name) {
        mName = name;
    }

    public Route(String name, LatLng startLoc, LatLng endLoc) {
        mName = name;
        mStartLoc = startLoc;
        mEndLoc = endLoc;
    }

    public Route(String name, LatLng startLoc, LatLng endLoc, String startName, String endName) {
        mName = name;
        mStartLoc = startLoc;
        mEndLoc = endLoc;
        mStartName = startName;
        mEndName = endName;
    }

    public Route(long id, int pos, String name, LatLng startLoc, LatLng endLoc, String startName, String endName) {
        mId = id;
        mPos = pos;
        mName = name;
        mStartLoc = startLoc;
        mEndLoc = endLoc;
        mStartName = startName;
        mEndName = endName;
    }

    long getId() {
        return mId;
    }
    void setId(long id) {
        mId = id;
    }

    int getPos() {
        return mPos;
    }
    void setPos(int mPos) {
        this.mPos = mPos;
    }

    public String getName() {
        return mName;
    }
    public void setName(String name) {
        mName = name;
    }

    LatLng getStartLoc() {
        return mStartLoc;
    }
    void setStartLoc(LatLng startLoc) {
        mStartLoc = startLoc;
    }

    LatLng getEndLoc() {
        return mEndLoc;
    }
    void setEndLoc(LatLng endLoc) {
        mEndLoc = endLoc;
    }

    String getStartName() {
        return mStartName;
    }
    void setStartName(String startName) {
        mStartName = startName;
    }

    String getEndName() {
        return mEndName;
    }
    void setEndName(String endName) {
        mEndName = endName;
    }

    float getApproximateDistance () {
        Location startLoc = new Location("start");
        startLoc.setLatitude(mStartLoc.latitude);
        startLoc.setLongitude(mStartLoc.longitude);
        Location endLoc = new Location("end");
        endLoc.setLatitude(mEndLoc.latitude);
        endLoc.setLongitude(mEndLoc.longitude);
        return startLoc.distanceTo(endLoc);
    }

    long getAverageWalkTime(Context context) {
        long time = 0;
        List<Walk> walks = DBManager.getWalks(context, this);
        if (walks.size() > 0) {
            for (Walk w : walks) {
                time += w.getDuration();
            }
            time /= walks.size();
        }
        return time;
    }
}

File: app/src/main/java/com/android/mathias/velocity/model/Route.kt
package com.android.mathias.velocity.model

import android.content.Context
import android.location.Location
import com.android.mathias.velocity.db.DBManager

import com.google.android.gms.maps.model.LatLng

internal class Route {
    var id: Long = 0
    var pos: Int = 0
    var name: String? = null
    var startLoc: LatLng? = null
    var endLoc: LatLng? = null
    var startName: String? = null
    var endName: String? = null

    val approximateDistance: Float
        get() {
            val startLoc = Location("start")
            startLoc.latitude = startLoc.latitude
            startLoc.longitude = startLoc.longitude
            val endLoc = Location("end")
            endLoc.latitude = endLoc.latitude
            endLoc.longitude = endLoc.longitude
            return startLoc.distanceTo(endLoc)
        }

    constructor()

    constructor(name: String) {
        this.name = name
    }

    constructor(name: String, startLoc: LatLng, endLoc: LatLng, startName: String, endName: String) {
        this.name = name
        this.startLoc = startLoc
        this.endLoc = endLoc
        this.startName = startName
        this.endName = endName
    }

    constructor(id: Long, pos: Int, name: String, startLoc: LatLng, endLoc: LatLng, startName: String, endName: String) {
        this.id = id
        this.pos = pos
        this.name = name
        this.startLoc = startLoc
        this.endLoc = endLoc
        this.startName = startName
        this.endName = endName
    }

    fun getAverageWalkTime(context: Context): Long {
        var time: Long = 0
        val walks = DBManager.getWalks(context, this)
        if (walks.isNotEmpty()) {
            for (w in walks) {
                time += w.duration
            }
            time /= walks.size.toLong()
        }
        return time
    }
}
